### ISSUE TITLE
Handshake - Decentralized Location for Casual Deal Making & Gentleman's Agreements

### ISSUE THREAD

**Issue by MCDForm** [district0x/district-proposals#158]
# Name
Handshake - Decentralized Location for Casual Deal Making & Gentleman's Agreements

# Purpose
To provide a decentralized location for virtual “Handshake Agreements” and to ultilize a commission producing blockchain transaction to timestamp the document.

A gentleman’s agreement is an arrangement or understanding which is based upon the trust of both or all parties.  It has been shown that a handshake agreement, whether written or verbal, is legally binding assuming the contract is valid.

The problem with these types of agreements is that there is often no proof they occurred.  This district will utilize trivial amounts small amounts of cryptocurrency and a blockchain based timestamp to provide proof of their existence.

A “Handshake” could replace more formal contracts or the need for attorneys, especially for “less important” agreements.  They could be used for small scale ventures, bets between friends, the sale of a digital item between two private parties and many other reasons.

A "Handshake" could also be used between friends just for fun but to put an agreement on record for eternity (or until the blockchain no longer exists).

# Method
When two parties enter an agreement there must be an offer and an acceptance.  There must also be consideration, which means that both parties agree to exchange something of actual value.

The district will provide an easy to use agreement form (similar to Madlibs). Party A will complete an agreement form, provided by the district, stating the offer and acceptance.  Party A will then send a small fee to the district while stating that they are completing the “Handshake”.  Party B will read the agreement and also send a small fee to the district signifying that they are also completing the “Handshake”.  The district will then include the two transaction numbers on the document providing a virtual timestamp.

# Case Use
<img src="https://user-images.githubusercontent.com/31578014/30199519-ee85eb3c-9427-11e7-8625-c9269d6e4e22.png" width="150"><img src="https://user-images.githubusercontent.com/31578014/30199656-84764bd2-9428-11e7-9125-9d608465b820.png" width="150"><img src="https://user-images.githubusercontent.com/31578014/30199686-a717b9a0-9428-11e7-8187-819d3d23df73.png" width="150">
<img src="https://user-images.githubusercontent.com/31578014/30199710-bf9b30ce-9428-11e7-8ccd-e39e075dccad.png" width="150"><img src="https://user-images.githubusercontent.com/31578014/30199732-d3fdb7bc-9428-11e7-90aa-efd002ba0a85.png" width="150">

The above images provide a walkthrough of a "handshake" via a cellphone.  The process is simple and can be completed in seconds.

The possibilities are endless for “Handshake” agreements.  Below are variations on how a Handshake could be used.

1. Bob bets his friend Doug 1 Eth that Prince William will have a fourth child.  They also agree that if the 4th child is a boy Bob gets an extra 1 Eth.  If it is a girl Doug will owe Bob 0 Eth.  They “Shake Hands” to keep the bet valid for eternity.

1.  You meet someone at the end of your backpacking trip and agree to let them borrow your wire frame backpack until they return to Los Angeles.  You “Shake Hands” and agree that you will be paid .5 Eth if the backpack is not returned by Dec 11th.

1.  Two friends start a business at their local college selling ice pops out of a cooler.  Lily and Tanya decide to be 50/50 partners but all of the equipment and supplies are Lily’s.  Lily and Tanya “Shake Hands” and agree that Tanya will work work four days a week to Lily’s three days a week until she has worked approximately 2 times as many hours.  If Tanya leaves the business she will pay Lily $200 minus $5 for each hour worked.

# Fee Structure
The fee that each party pays to enter the agreement will be used to cover any transaction fees and profits will be split amongst the District stakeholders.  

If a single party extends their hand to the other (pays their half of the fee) but the second half does not reciprocate the handshake the initial fee is captured by the district and not refunded.

# Future Growth
1.  In the future it may be possible to offer "handshakes" with a smart contract component.  The smart contract can be used to hold tokens until the conditions of the agreement have been met.
1.  Incorporating money and token transfer could add an additional money stream.   
1.  Group handshakes - Handshakes with more than two parties

0x1c63a09C62558B2A614f48e7495291f02f99AC40

**Comment by basiccrypto** [comment]
Very straight forward use case with some large utility gains, especially when time is short and a more convenient option exists.  Not that this is directly related, but got me thinking about this article with processing time of banks vs cryptocurrency: https://cointelegraph.com/news/how-bitcoin-rescued-victims-of-hurricane-harvey

I like this, but seems pretty similar to #67 and #97 in some aspects, how would this differ against those?

I think one key issue is having some form of validation of the outcome, whether that be Oracle's, Smart Contracts based on time, Multi-sig resolution, or a 3rd party, some mechanism is needed to ensure fairness.

**Comment by mckmuze** [comment]
Can I get a token address for this proposal please

**Comment by MCDForm** [comment]
0x1c63a09C62558B2A614f48e7495291f02f99AC40

Here you go.

**Comment by MCDForm** [comment]
@basiccrypto The main difference between this and #97 is that it is not strictly a betting district.  In the past many huge deals were made over a handshake (McDonalds - Not 100% true but you get the point).  I see this district mostly as being used between friends or the more laid back crypto set in a casual way without the need for 3rd party resolution or escrow. Butjust in case two friends decide to start a burger place while having a beer and agree to split the business  50/50 this will give them the chance to put that deal in writing (in seconds) and maybe save one of them from getting erased from the records in the future.

However, I included escrow/smart contracts (based on your comment) in my updated proposal because I could see the users eventually wanting it.  If the need for smart contracts and escrow accounts is ever realized then I could see this district using a service like #67 more than being one.

**Comment by imdying** [comment]
How is this superior to an email between friends?

**Comment by Bradymck** [comment]
I love how I have a hard time classifying your proposals. True signifier that it is an original idea.

**Comment by drummingjack2** [comment]
@MCDForm have you thought gut about the data source for these handshakes? In your example of whether or not someone has a forth child, would there be some form of external “Oracle”, a third party that signals when the event does (or does not) take place?

Or is this left up to the participants of the handshake to verify, on the honor system? I worked on proposal #97 and had a similar issue with external third party identification.

**Comment by drummingjack2** [comment]
fourth*

**Comment by MCDForm** [comment]
@drummingjack2 In this proposal it is more on the honor system.  It's a gateway to blockchain as proof.  I could see something like Tierion sending data about the fourth child though.. 

It's going to be very hard to translate real life actions into events recognized by smart contracts.  For some of your examples I could see the IoT playing a part.  Can I lose 10 lbs by January 1st?  Maybe a scale can be tied to an oracle.

**Comment by Bradymck** [comment]
@MCDForm I agree, it is challenging but some centralized apps do just as you describe. We could implement similar models. I think the app is called Dietbet IIRC
